### PR TITLE
Feat: `Skeleton` component

### PR DESCRIPTION
**Pull request by HamidG420**

### CODE REVIEW

File: src/skeleton/skeleton.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { html, LitElement, nothing } from 'lit';
+import { property } from 'lit/decorators.js';
+import { SkeletonAnimation, SkeletonVariant } from './types';
+
+export class Skeleton extends LitElement {
+  @property({ reflect: true }) variant?: SkeletonVariant = 'line';
+  @property({ reflect: true }) animationMode?: SkeletonAnimation = 'progress';
```
Comment (amir78729) [inline src/skeleton/skeleton.ts:7]:
```suggestion
  @property({ reflect: true, attribute: 'animation-mode' }) animationMode?: SkeletonAnimation = 'progress';
```

File: src/skeleton/index.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+ * ```html
+ * <tap-skeleton animationMode="progress"></tap-skeleton>
+ * <tap-skeleton animationMode="pulse"></tap-skeleton>
+ * <tap-skeleton animationMode="none"></tap-skeleton>
+ * ```
+ *
+ * @summary Display Skeleton component with different styles and types.
+ *
+ * @prop {`"line"` \| `"rect"` \| `"circle"`} [`variant`=`"line"`] - The variant of the skeleton.
+ * @prop {`"progress"` \| `"pulse"` \| `"none"`} [`animationMode`=`"progress"`] - The animation mode of the skeleton.
```
Comment (amir78729) [inline src/skeleton/index.ts:33]:
```suggestion
 * @prop {`"progress"` \| `"pulse"` \| `"none"`} [`animation-mode`=`"progress"`] - The animation mode of the skeleton.
```

File: src/skeleton/index.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+ * <tap-skeleton variant="rect"></tap-skeleton>
+ * <tap-skeleton variant="circle"></tap-skeleton>
+ * ```
+ *
+ * ##### Animation Modes
+ *
+ * ```html
+ * <tap-skeleton animationMode="progress"></tap-skeleton>
+ * <tap-skeleton animationMode="pulse"></tap-skeleton>
+ * <tap-skeleton animationMode="none"></tap-skeleton>
```
Comment (amir78729) [inline src/skeleton/index.ts:27]:
```suggestion
 * <tap-skeleton animation-mode="progress"></tap-skeleton>
 * <tap-skeleton animation-mode="pulse"></tap-skeleton>
 * <tap-skeleton animation-mode="none"></tap-skeleton>
```

File: src/skeleton/skeleton.style.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+  :host([variant='circle']) .skeleton {
+    width: var(--tap-skeleton-circle-width, var(--tap-sys-spacing-10));
+    height: var(--tap-skeleton-circle-height, var(--tap-sys-spacing-10));
+    border-radius: var(
+      --tap-skeleton-circle-radius,
+      var(--tap-sys-radius-full)
+    );
+  }
+
+  :host([animationMode='progress']) .skeleton {
```
Comment (amir78729) [inline src/skeleton/skeleton.style.ts:47]:
```suggestion
  :host([animation-mode='progress']) .skeleton {
```

File: src/skeleton/skeleton.style.ts [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    animation: progress 2s cubic-bezier(0.4, 0, 0.2, 1) infinite;
+    background-size: 200px 100%;
+    background-image: linear-gradient(
+      90deg,
+      color-mix(in srgb, var(--tap-sys-color-surface-white), transparent 100%),
+      color-mix(in srgb, var(--tap-sys-color-surface-white), transparent 40%),
+      color-mix(in srgb, var(--tap-sys-color-surface-white), transparent 100%)
+    );
+  }
+  :host([animationMode='pulse']) .skeleton {
```
Comment (amir78729) [inline src/skeleton/skeleton.style.ts:57]:
```suggestion
  :host([animation-mode='pulse']) .skeleton {
```

File: src/skeleton/skeleton.style.ts [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+    100% {
+      opacity: 1;
+    }
+  }
+
+  // Removing animation if user enabled the 'Reduce Motion' option.
+  @media (prefers-reduced-motion: reduce) {
+    :host([animationMode='progress']) .skeleton,
+    :host([animationMode='pulse']) .skeleton {
```
Comment (amir78729) [inline src/skeleton/skeleton.style.ts:87]:
```suggestion
    :host([animation-mode='progress']) .skeleton,
    :host([animation-mode='pulse']) .skeleton {
```

**Comment by HamidG420** [comment]
I have addressed the comments and resolved them.

**Comment by amir78729** [comment]
@HamidG420 LGTM ✨ 
@majidsajadi Please check this PR

File: src/skeleton/skeleton.ts [hunk 7]
```diff
@@ -0,0 +1,22 @@
+import { html, LitElement, nothing } from 'lit';
+import { property } from 'lit/decorators.js';
+import { SkeletonAnimation, SkeletonVariant } from './types';
+
+export class Skeleton extends LitElement {
+  @property({ reflect: true }) variant?: SkeletonVariant = 'line';
+  @property({ reflect: true, attribute: 'animation-mode' })
+  animationMode?: SkeletonAnimation = 'progress';
```
Comment (majidsajadi) [inline src/skeleton/skeleton.ts:8]: The animations are not defined in design system but i think we can merge this and change the animation according the design system in future.

File: src/skeleton/types.ts [hunk 8]
```diff
@@ -0,0 +1,2 @@
+export type SkeletonAnimation = 'progress' | 'none' | 'pulse';
+export type SkeletonVariant = 'line' | 'rect' | 'circle';
```
Comment (majidsajadi) [inline src/skeleton/types.ts:2]: The design system variants are different. but I think those variants can be computed using these. 
We can go with this state and change it if its not flexible enough.

**Comment by majidsajadi** [comment]
@HamidG420  LGTM. Thanks.

**Pull request merged**